libpod/runtime.go

```diff
@@ -93,50 +158,140 @@ var (
-	return pathHelper(paths)
-}
+// NewRuntime creates a new container runtime
+// Options can be passed to override the default configuration for the runtime
+func NewRuntime(options ...RuntimeOption) (runtime *Runtime, err error) {
+	runtime = new(Runtime)
+	runtime.config = new(RuntimeConfig)
 
-func findRuncPath() string {
-	paths := []string{"/usr/bin/runc", "/usr/sbin/runc", "/sbin/runc", "/bin/runc", "/usr/lib/cri-o-runc/sbin/runc"}
-	return pathHelper(paths)
-}
+	// Copy the default configuration
+	deepcopier.Copy(defaultRuntimeConfig).To(runtime.config)
+
+	configPath := ConfigPath
+	foundConfig := true
+	if _, err := os.Stat(OverrideConfigPath); err == nil {
+		// Use the override configuration path
+		configPath = OverrideConfigPath
+	} else if _, err := os.Stat(ConfigPath); err != nil {
+		// Both stat checks failed, no config found
+		foundConfig = false
+	}
 
-func pathHelper(paths []string) string {
-	for _, path := range paths {
-		if _, err := os.Stat(path); err == nil {
-			return path
+	// If we have a valid configuration file, load it in
+	if foundConfig {
+		contents, err := ioutil.ReadFile(configPath)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error reading configuration file %s", configPath)
+		}
+		if _, err := toml.Decode(string(contents), runtime.config); err != nil {
+			return nil, errors.Wrapf(err, "error decoding configuration file %s", configPath)
 		}
 	}
-	return paths[0]
+
+	// Overwrite config with user-given configuration options
+	for _, opt := range options {
+		if err := opt(runtime); err != nil {
+			return nil, errors.Wrapf(err, "error configuring runtime")
+		}
+	}
+
+	if err := makeRuntime(runtime); err != nil {
+		return nil, err
+	}
+
+	return runtime, nil
 }
 
-// NewRuntime creates a new container runtime
-// Options can be passed to override the default configuration for the runtime
-func NewRuntime(options ...RuntimeOption) (runtime *Runtime, err error) {
+// NewRuntimeFromConfig creates a new container runtime using the given
+// configuration file for its default configuration. Passed RuntimeOption
+// functions can be used to mutate this configuration further.
+// An error will be returned if the configuration file at the given path does
+// not exist or cannot be loaded
+func NewRuntimeFromConfig(configPath string, options ...RuntimeOption) (runtime *Runtime, err error) {
 	runtime = new(Runtime)
 	runtime.config = new(RuntimeConfig)
 
-	// Copy the default configuration
-	deepcopier.Copy(defaultRuntimeConfig).To(runtime.config)
+	// Set two fields not in the TOML config
+	runtime.config.StateType = defaultRuntimeConfig.StateType
+	runtime.config.StorageConfig = storage.StoreOptions{}
 
-	// Overwrite it with user-given configuration options
+	// Check to see if the given configuration file exists
+	if _, err := os.Stat(configPath); err != nil {
+		return nil, errors.Wrapf(err, "error stating configuration file %s", configPath)
```

Comment:
might be me, but I'd pref getting rid of 'stating'.  Perhaps:

"error performing stat on configuration file"
"error unable to determine status of configuration file"

When I saw stating, my first thought is you had trouble starting the config file.